gdal/ogr/ogrsf_frmts/shape/dbfopen.c

```diff
@@ -304,16 +304,16 @@ void SHPAPI_CALL
 DBFUpdateHeader( DBFHandle psDBF )
 
 {
-    unsigned char		abyFileHeader[XBASE_FILEHDR_SZ];
 
     if( psDBF->bNoHeader )
         DBFWriteHeader( psDBF );
 
     if( !DBFFlushRecord( psDBF ) )
         return;
 
+    unsigned char abyFileHeader[XBASE_FILEHDR_SZ] = {0}; 
```

Comment:
when merged back in shapelib, this is still intended to be C89 I think, so the declaration must be kept at the beginning

Reply:
Really? :(. Is there any way to get it to C99?  20 years ago versus 30 years ago... I'll make an issue https://github.com/OSGeo/shapelib for discussion.

Reply:
I'm afraid I don't have any motivation/time/energy to discuss language updates for shapelib, for which I'm already a bad maintainer, especially for a one line change that can be easily avoided.

Reply:
Moved declaration back up.